chore: add python import sorting

sfoster1 opened the pull request:
Using ruff's https://docs.astral.sh/ruff/settings/#lintisort isort integration we can sort our python imports! Finally!

The first commit has the settings for this, the rest of the commits just add them.

Draft for circular import reasons.

sfoster1 committed:
chore: sort python imports

Add ruff isort integration!

sfoster1 committed:
chore(api): sort imports

sfoster1 committed:
chore(g-code-testing): sort imports

sfoster1 committed:
chore(hardware): sort imports

sfoster1 committed:
chore(robot-server): sort imports

sfoster1 committed:
chore(shared-data): sort imports

sfoster1 committed:
chore(system-server): sort imports

sfoster1 committed:
chore(update-server): sort imports

sfoster1 committed:
chore(usb-bridge): sort imports

sfoster1 committed:
chore(system-server): sort imports

ddcc4 reviewed (approved):
Wow, this is a big diff. I'm glad we don't have an active release branch that will create merge conflicts with this :)

SyntaxColoring reviewed (approved):
LGTM when the circular import stuff is resolved (condolences).

sfoster1 committed:
chore(hardware): one last twiddle

sfoster1 committed:
chore: an exemplar circular import fix

There are others, but this is worth looking at independently. We have,
in general, a lot of bad situations with opentrons.config. One really
bad one is that opentrons.config imports opentrons.hardware_control,
essentially only for OT3AxisType, and then hardware control imports
config, like, everywhere.

We can fix this by defining OT3AxisType in config - it's really intended
for config - but we need to be aware that it's not realllly the same
separation of concerns as before.

sfoster1 committed:
chore(api): more circular imports

Lots of weird little bits and bobs about making things deferred or type
checking only imports. it all works.

sfoster1 committed:
chore(robot-server): noqa after import resort

sfoster1 committed:
update: vacuum module after rebase

SyntaxColoring reviewed (approved):
Chaos bows to order.

sfoster1 committed:
ignore generated file

sfoster1 committed:
ignore other generated files

sfoster1 committed:
format
